CONTRIBUTION-GUIDE.md

```diff
@@ -0,0 +1,151 @@
+# Contribution Guide
+
+Thank you for your interest in contributing to this project! We welcome contributions from the community and appreciate your help in improving the project. We highly encourage meaningful pull requests (PRs) that include bug fixes, performance optimizations, or functional improvements. Additionally, opening issues to report critical bugs or suggest valuable improvements is just as important as submitting PRs.
+
+*Because maintainers need to sign your commits, PRs with typo fixes are generally not favorable. Such PRs will be closed and replaced with an issue instead.*
+
+## Repository Structure
+
+Understanding the project structure will help you navigate and contribute effectively. Below are the key directories and their purposes:
+
+- **`nil/`** – Contains the core blockchain logic, including consensus mechanisms, networking protocols, cryptographic primitives, and state management functionalities.
+
+- **`clijs/`** – Houses the command-line interface tools for interacting with the nil blockchain, facilitating tasks such as node management, transaction submission, and network monitoring.
+
+- **`create-nil-hardhat-project/`** – Provides a template or scaffolding for developers to quickly set up a Hardhat project tailored for the nil blockchain, streamlining the development and deployment of smart contracts.
+
+- **`docs/`** – Contains comprehensive documentation, including protocol specifications, developer guides, API references, and integration tutorials.
+
+- **`explorer_backend/`** – Manages the backend services for the blockchain explorer, including data indexing, API endpoints, and real-time data processing.
+
+- **`explorer_frontend/`** – Develops the user interface for the blockchain explorer, enabling users to visualize blockchain data, track transactions, and monitor network status.
+
+- **`niljs/`** – A TypeScript client library providing tools and abstractions for interacting with the nil blockchain.
```

Comment:
let's put this one first, after nil. I mean let's organize this list by priority